src/db/migrate.rs

```diff
@@ -502,6 +502,33 @@ pub fn migrate(version: Option<Version>, conn: &mut Client) -> CratesfyiResult<(
             "
                  ALTER TYPE feature DROP ATTRIBUTE optional_dependency;
             "
+        ),
+        migration!(
+            context,
+            22,
+            // description
+            "Add the github_repositories table to contain GitHub information",
+            // upgrade query
+            "
+                CREATE TABLE github_repos (
+                    id VARCHAR PRIMARY KEY NOT NULL,
+                    name VARCHAR NOT NULL,
+                    description VARCHAR,
+                    last_commit TIMESTAMP,
+                    stars INT NOT NULL,
+                    forks INT NOT NULL,
+                    issues INT NOT NULL,
+                    updated_at TIMESTAMP NOT NULL
+                );
+
+                ALTER TABLE releases ADD COLUMN github_repo VARCHAR
+                    REFERENCES github_repos(id) ON DELETE SET NULL;
```

Comment:
This will be NULL by default, and then set to the right repository as github_updater runs? That doesn't have a way to distinguish between 'not yet updated' and 'deleted repo', will we need to know that? Or is the plan to have everything converted before switching this off?

In other words, I think this doesn't give us a way to know if the updater has run on a given repo, so we don't know whether we need to let it run longer.

Reply:
My plan is to have the updater go through every repository before removing the "legacy" code. For crates published after the switch to GraphQL, my plan is to gather their GitHub stats at the end of the build process.

Reply:
> My plan is to have the updater go through every repository before removing the "legacy" code. For crates published after the switch to GraphQL, my plan is to gather their GitHub stats at the end of the build process.

Right, but how do you tell when it's gone through every repository? You can't tell by looking at the database because it's valid normally for the `github_repo` to be NULL.

Reply:
> Right, but how do you tell when it's gone through every repository?

It goes through every repository each day. The number of crates that weren't updated in a day (query 1) is just 1369, and when randomly checking their repositories (clicking random links of query 2) all of them were 404s. If we leave it running over the weekend we'll surely convert all of the crates to the new schema.

```sql
-- Query 1
SELECT COUNT(*) FROM crates WHERE github_last_update < NOW() - INTERVAL '1 day 1 hour';
-- Query 2:
SELECT DISTINCT (crates.id) crates.name, releases.repository_url FROM crates, releases WHERE crates.id = releases.crate_id AND github_last_update < NOW() - INTERVAL '1 day 1 hour';
```

The only thing we'd be missing is newly published crates, but I'll handle them in a followup PR.

Reply:
Ok, seems reasonable.